src/foam/nanos/client/ClientBuilder.js

```diff
@@ -38,98 +39,65 @@ foam.CLASS({
           })
         });
       }
-    }
-  ],
-
-  methods: [
-    function then(resolve) {
-      var self = this;
-
-      var client = {
-        package: 'foam.nanos.client',
-        name: 'Client',
-
-        implements: [ 'foam.box.Context' ],
-
-        requires: [
-          'foam.box.HTTPBox',
-          'foam.dao.RequestResponseClientDAO',
-          'foam.dao.ClientDAO',
-          'foam.dao.EasyDAO'
-        ],
-
-        exports: [
-        ],
-
-        properties: [
-        ],
-
-        methods: [
-          function createDAO(config) {
-            config.daoType = 'MDAO'; // 'IDB';
-            config.cache   = true;
-
-            return this.EasyDAO.create(config);
-          }
-        ]
-      };
-
-      var references = [];
-
-      // Force hard reload when app version updates
-      self.nSpecDAO.find("appConfig").then(function(spec) {
-        var appConfig = spec.service;
-        var version   = appConfig.version;
-
-        if ( "CLIENT_VERSION" in localStorage ) {
-          var oldVersion = localStorage.CLIENT_VERSION;
-          if ( version != oldVersion ) {
-            localStorage.CLIENT_VERSION = version;
-            location.reload(true);
-          }
-        } else {
-          localStorage.CLIENT_VERSION = version;
-        }
-      });
+    },
+    {
+      name: 'promise',
+      factory: function() {
+        var self = this;
+        return new Promise(function(resolve) {
+          var client = {
+            package: 'foam.nanos.client',
```

Comment:
I still think that ClientBuilder should be building just a Context and not a Class, otherwise we can't talk to more than one nanos server at time.

But I guess we can leave that as a TODO for now.

Reply:
But add a TODO for it.